src/services/addressProvider.ts

```diff
@@ -40,15 +48,44 @@ export class AddressProvider<T extends ChainId> extends Service {
       case 42161:
         return "0xcAd10033C86B0C1ED6bfcCAa2FF6779938558E9f";
       default:
-        throw new Error("Unsupported chain id");
+        throw new Error(`Unsupported chain id: ${chainId}`);
     }
   }
 
-  async addressById(id: ContractAddressId): Promise<string> {
+  async addressById(id: ContractAddressId): Promise<Address> {
+    const { address, timestamp } = this.cachedAddressesById.get(id) || {};
+
+    if (address && this.isCacheFresh({ timestamp })) {
+      return address;
+    }
+
     return this.contract.read.addressById(id);
```

Comment:
should we be setting the cache here? So we don't have to do it e.g. in the vaults interface/where we use it?

so I guess it'd be:
```
const address = await this.contract.read.addressById(id);
this.setCachedAddressById(id, address);
return address
```

Reply:
I thought there was a reason we were awaiting in the vaults interface/where we use the `addressById` and this `AddressProvider#addressById()` was suppose to return a promise to be awaited elsewhere?

Returning just the `Address` without the other places having to await for it it's definitely cleaner, but surely there was a reason we were doing it this way, no?

Reply:
Done with https://github.com/yearn/yearn-sdk/pull/279/commits/1fa4d16681b27c6b30fd1dbb3f45aa237c77f636